app/services/sso/jwt.rb

```diff
@@ -8,12 +9,11 @@ def available?
 
     def authenticate!
       payload = jwt_token.decode || {}
-      user_id = payload['user_id']
-      user = User.unscoped.except_hidden.find_by(id: user_id) if user_id
-      @current_user = user
-      user&.login
+      store(payload)
```

Comment:
I believe this should go into a separate SSO method. Something like `SSO::OpenIdConnect`. As I said before, instead of implementing this yourself, why not use a gem for this?

Reply:
@timogoebel : The only thing that stops me from doing that is I recieve the access token in the form 'Bearer {Token}'. If I receive the token in this fashion then jwt.rb with be set as `available_sso` and then it will call its methods accordingly. Can you suggest me a way to bypass this?

Reply:
You can tweak the algorithm so that the sso is just available if e.h. `user_id` is present in the JWT data.

Reply:
We could also change the header for JWTs as we haven't released it. `Bearer` is basically an arbitrary string that specifies the type of authentication. The current JWT implementation is not using Oauth 2.0, so you could consider `Bearer` to be wrong. Let's switch it to `JWT` and we're good. See https://stackoverflow.com/questions/7802116/custom-http-authorization-header

Reply:
@tbrisker Yes, so in `jwt.rb` we are using 'JWT' as a prefix to the token, whereas in the `openid_connect.rb` we are using the 'Bearer' as a prefix.